### PR TITLE
Funcionality to toggle the entire table added

### PR DESCRIPTION
**Pull request by likefurnis**
- Done the same way it has been implemented on `p-panel`
- Added CSS from `p-panel and` some custom CSS to keep the icon at the right of the caption
- Working and ready to test on `tabledemo.html`

References issue  #9463

### CODE REVIEW

File: src/app/components/table/table.css [hunk 1]
```diff
@@ -250,6 +250,31 @@ p-columnfilterformelement > .p-inputtext,
...
+.p-datatable-header-icon {
+    display: inline-flex;
+    justify-content: center;
+    align-items: center;
+    cursor: pointer;
+    text-decoration: none;
+    overflow: hidden;
+    position: relative;
+}
+
```
Comment (likefurnis) [inline src/app/components/table/table.css:277]: Styles imported from `p-panel`

File: src/app/components/table/table.css [hunk 2]
```diff
@@ -250,6 +250,31 @@ p-columnfilterformelement > .p-inputtext,
...
+.p-datatable-header-content {
+    flex: 1;
+}
+
+.p-datatable-header-icons {
+    margin-left: 1rem;
+}
+
```
Comment (likefurnis) [inline src/app/components/table/table.css:267]: Custom styles to have the icon on the caption always on the right.

File: src/app/showcase/components/table/tabledemo.html [hunk 3]
```diff
@@ -10,7 +10,7 @@ <h1>Table</h1>
         <p-table #dt [value]="customers" [(selection)]="selectedCustomers" dataKey="id" styleClass="p-datatable-customers" [rowHover]="true"
             [rows]="10" [showCurrentPageReport]="true" [rowsPerPageOptions]="[10,25,50]" [loading]="loading"
             [paginator]="true" currentPageReportTemplate="Showing {first} to {last} of {totalRecords} entries"
-            [filterDelay]="0" [globalFilterFields]="['name','country.name','representative.name','status']">
+            [filterDelay]="0" [globalFilterFields]="['name','country.name','representative.name','status']" [toggleable]="true">
```
Comment (likefurnis) [inline src/app/showcase/components/table/tabledemo.html:13]: Example on `tabledemo.html` working.

**Comment by cagataycivici** [comment]
Thank you for the PR however due to size of the changes, we're unable to accept it. We may put up a demo or create a directive to implement such feature in the future.

**Pull request closed without merging**